### DISCUSSION TITLE
"unsigned" not implemented ?

### DISCUSSION THREAD

**Discussion by ttttttom** [bxparks/AUnit#89] [Q&A] [answered]
The following program produces a compile error: 
`error: call of overloaded 'assertion(const char [15], int, unsigned int&, const char [3], <unresolved overloaded function type>, int)' is ambiguous`

```Arduino
#line 2 "test_aunit.ino"
#include <AUnit.h>

void setup()
{
  Serial.begin(115200);
}

void loop()
{
  aunit::TestRunner::run();
}

test (1)
{
  unsigned int i = 1; // Makro-Fehler bei Verwendung von "unsigned"
  assertEqual(i, 1);
}
```

**Comment by bxparks** [comment, marked as answer]
The 2 arguments to the `assertEqual()` must match **exactly** in AUnit. See https://github.com/bxparks/AUnit#ParameterTypesMustMatch which explains this.
